lib/identity_tijuana.rb

```diff
@@ -173,38 +173,47 @@ def self.fetch_donation_updates(sync_id)
 
   def self.fetch_donation_updates_impl(sync_id)
     started_at = DateTime.now
-    last_updated_at = get_redis_date('tijuana:donations:last_updated_at')
-    last_id = (Sidekiq.redis { |r| r.get 'tijuana:donations:last_id' } || 0).to_i
+    last_updated_at = get_redis_date('tijuana:transactions:last_updated_at')
+    last_id = (Sidekiq.redis { |r| r.get 'tijuana:transactions:last_id' } || 0).to_i
     users_dependent_data_cutoff = get_redis_date('tijuana:users:dependent_data_cutoff')
-    updated_donations = IdentityTijuana::Donation.updated_donations(last_updated_at, last_id, users_dependent_data_cutoff)
-    updated_donations_all = IdentityTijuana::Donation.updated_donations_all(last_updated_at, last_id, users_dependent_data_cutoff)
-    updated_donations.each do |donation|
-      IdentityTijuana::Donation.import(donation.id, sync_id)
+
+    updated_transactions = IdentityTijuana::Transaction.updated_transactions(
+      last_updated_at, last_id, users_dependent_data_cutoff
+    )
+
+    updated_transactions_all = IdentityTijuana::Transaction.updated_transactions_all(
+      last_updated_at, last_id, users_dependent_data_cutoff
+    ).count('id')
+
+    updated_donation_ids = updated_transactions.pluck(:donation_id).compact.uniq
+
+    updated_donation_ids.each do |donation_id|
+      IdentityTijuana::Donation.import(donation_id, sync_id)
```

Comment:
I know you didn't write this call, but maybe worth investigating why this approach here and the subsequent lookup in `Donation::import` used.

In ActiveRecord entity identity is tied to it id value, so it should be possible to work with the `Transaction.donation` relation directly, rather than getting the id and subsequently having to do a `Donations.find()` to load each one individually - that is always a massive perf hit. Instead, the donations could be loaded along with the transactions in the same SQL query using [eager loading](https://guides.rubyonrails.org/active_record_querying.html#eager-loading-associations).

Or even perhaps better inverting how this works by loading all distinct donations with matching Transaction ids, and letting the DB handle finding distinct donations and loading them all in one hit that way.

Reply:
Yes, had the same, or at least a similar, thought, but again just didn't want to change the approach.

Not sure how exactly you mean `distinct` donations with matching Transaction ids – not sure that loading Transactions Ids first, then finding distinct Donations is the most optimal. 
The point here is that it is the transactions that are being created, whilst donations are a bit less frequently created/modified (except for the one-offs). So loading donations even eagerly with transactions leads to potentially processing same transactions over again.

What I wanted to do was to eagerly load transactions with their donation, then provide both in the `import` payload. That way we get everything necessary loaded in one call, and the `import` doesn't need to loop through all donation's transactions, but only through the actually modified ones.

Reply:
Yeah I hear you. Think I'm pretty happy with the updated approach tbh.